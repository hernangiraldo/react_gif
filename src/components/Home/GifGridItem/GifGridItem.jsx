import React from 'react';
import PropTypes from 'prop-types';

import './GifGridItem.scss';

const GifGridItem = ({ title, url }) => {
  return(
    <div className='GifGridItem animate__animated animate__fadeIn'>
      <h3>{ title }</h3>
      <img src={ url } alt={ title }/>
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGridItem;
