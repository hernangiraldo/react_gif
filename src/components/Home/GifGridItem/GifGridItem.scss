.GifGridItem {
  padding: 1rem;
}
