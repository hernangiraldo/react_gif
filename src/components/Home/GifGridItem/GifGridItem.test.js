import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from './GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

  const title = 'Mi tÃ­tulo';
  const url = 'https://localhost:3000'

  test('Debe mostrar el componente correctamente', () => {

    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    expect(wrapper).toMatchSnapshot();

  })

})
