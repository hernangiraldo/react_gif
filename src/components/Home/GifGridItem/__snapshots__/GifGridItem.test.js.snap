// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="GifGridItem animate__animated animate__fadeIn"
>
  <h3>
    Mi título
  </h3>
  <img
    alt="Mi título"
    src="https://localhost:3000"
  />
</div>
`;
